### ISSUE TITLE
each_pair deprecation warning (6.x)

### ISSUE THREAD

**Issue by conorom** [projectblacklight/blacklight#1814]
For several months we've been getting a deprecation warning from a spec that commits to Solr.
It's not a big deal but I'm tired of seeing it ;-)

Looking at the closed issues, I think the problem is very similar to this:
https://github.com/projectblacklight/blacklight/issues/1757

And can be solved by adding `each_pair` here:
https://github.com/projectblacklight/blacklight/blob/c06fbd6535e82b8bae0087dc1e3d252ca1c7e18b/app/models/concerns/blacklight/document.rb#L72

```
==== DEPRECATION WARNINGS ====
DEPRECATION WARNING: Blacklight::Document#each_pair is deprecated; use obj.to_h.each_pair instead. (called from initialize at /Users/conorom/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/rsolr-2.0.2/lib/rsolr/document.rb:14) (116 times); e.g.:
....
....
```

**Comment by jcoyne** [comment]
@conorom Do you still get this message if you update rsolr to 2.1.0?

**Comment by conorom** [comment]
Yup. The usage of `each_pair` is still there in rsolr 2.1.0 and continues causing the Blacklight deprecation warning:
https://github.com/rsolr/rsolr/blob/17b343cf56b44f257c2c06edaf01b39f10c9cbe2/lib/rsolr/document.rb#L14

**Comment by jcoyne** [comment]
Okay, so it looks to me like the problem is that Blacklight is passing a `SolrDocument` as the argument to `RSolr::Document.new`, but it should instead be passing a hash.

**Comment by jcoyne** [comment]
Do you have a full backtrace @conorom? I don't understand where Blacklight is creating this `RSolr::Document`

**Comment by conorom** [comment]
After spending some time doing battle with the deprecation gem to get a backtrace including Blacklight (and failing) I thought about what you are saying and it rang a little bell.

Indeed, to my chagrin, the darn test is calling `ActiveFedora::SolrService.add` with SolrDocuments instead of hashes, when the method clearly states what it wants:
https://github.com/samvera/active_fedora/blob/03d001bd79dcdfd1d85a95d1472e8bc3c4df9a93/lib/active_fedora/solr_service.rb#L86
The monstrous spec, FYI:
https://github.com/mlibrary/heliotrope/blob/8716c3098e964bfa921af0248b562ddaf949b909/spec/presenters/hyrax/monograph_presenter_spec.rb#L47

Worse still, I think this fix crossed my mind before but when I went to change the spec I started creating hashes from the get-go, which broke more of the spec than I had time to deal with. Safe to say at some later point than that I googled the deprecation warning, found your issue and thought that might be it. Bah! Sorry for the wild goose chase :(

Aaanyway, converting the docs to hashes just before adding to Solr fixes this extremely easily. I really appreciate the nudge back in the right direction!
